### PR TITLE
Extract Indexer class and start in parallel

### PR DESCRIPTION
**Pull request by philandstuff**
Because the DestinationDBUpdateDAO instance tries to create an index in
its constructor, and we create these objects in series in the main()
method, we can block the indexing of all registers while we try to
create an expensive index for a single big register.

This extracts that code into a new Indexer class, and starts these
indexers in parallel, so that we can start consuming one register while
creating a GIN index on another.

Once this is merged, we can revert #19.

### CODE REVIEW

File: src/main/java/uk/gov/indexer/Application.java [hunk 1]
```diff
@@ -1,87 +1,32 @@
...
-
-    private static void addShutdownHook(final ScheduledExecutorService executorService) {
-        runtime.addShutdownHook(new Thread() {
+    private static void addShutdownHook(final ScheduledExecutorService executorService, Stream<Indexer> indexerStream) {
+        Runtime.getRuntime().addShutdownHook(new Thread() {
             @Override
             public void run() {
                 executorService.shutdown();
-                databaseObjectRegistry.forEach(DBConnectionDAO::close);
+                indexerStream.forEach(Indexer::stop);
```
Comment (om-sharma) [inline src/main/java/uk/gov/indexer/Application.java]: Can you reuse the Stream? This statement will cause in error because the stream is already opened.

Comment (philandstuff) [inline src/main/java/uk/gov/indexer/Application.java]: :( good point

Comment (om-sharma) [inline src/main/java/uk/gov/indexer/Application.java]: Fixed the issue in next commit and merging the PR.

**Comment by philandstuff** [comment]
ta :+1:

**Pull request merged**